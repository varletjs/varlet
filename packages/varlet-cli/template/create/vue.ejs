<template>
  <div class="var-<%- kebabCaseName %>"></div>
</template>

<script lang="ts">
import { defineComponent } from 'vue'
import './<%- kebabCaseName %>.less'

export default defineComponent({
  name: '<%- namespace + bigCamelizeName %>',
})
</script>
