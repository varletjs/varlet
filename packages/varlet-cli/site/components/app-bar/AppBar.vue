<template>
  <div
    class="var-site-app-bar"
    :class="{ 'var-site-elevation--3': elevation }"
    :style="{
      background: color,
      color: textColor,
    }"
  >
    <div class="var-site-app-bar__left">
      <slot name="left" />
      <div
        class="var-site-app-bar__title"
        :style="{ paddingLeft }"
        v-if="titlePosition === 'left'"
      >
        <slot>{{ title }}</slot>
      </div>
    </div>

    <div class="var-site-app-bar__title" v-if="titlePosition === 'center'">
      <slot>{{ title }}</slot>
    </div>

    <div class="var-site-app-bar__right">
      <div
        class="var-site-app-bar__title"
        :style="{ paddingRight }"
        v-if="titlePosition === 'right'"
      >
        <slot>{{ title }}</slot>
      </div>
      <slot name="right" />
    </div>
  </div>
</template>

<script lang="ts">
import {defineComponent, onMounted, onUpdated, ref, Ref} from 'vue'
import { props } from './props'

export default defineComponent({
  name: 'VarSiteAppBar',
  props,
  setup(props, { slots }) {
    const paddingLeft: Ref<number | undefined> = ref()
    const paddingRight: Ref<number | undefined> = ref()

    const computePadding = () => {
      paddingLeft.value = slots.left ? 0 : undefined
      paddingRight.value = slots.right ? 0 : undefined
    }

    onMounted(computePadding)
    onUpdated(computePadding)

    return {
      paddingLeft,
      paddingRight
    }
  }
})
</script>

<style lang="less">
@import '../styles/common';
@import '../styles/elevation';
@import './appBar';
</style>
