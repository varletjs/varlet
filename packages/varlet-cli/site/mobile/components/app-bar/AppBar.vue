<template>
  <div
    :class="classes(n(), [elevation, 'var-site-elevation--3'])"
    :style="{
      background: color,
      color: textColor,
    }"
  >
    <div :class="n('left')">
      <slot name="left" />
      <div :class="n('title')" :style="{ paddingLeft }" v-if="titlePosition === 'left'">
        <slot>{{ title }}</slot>
      </div>
    </div>

    <div :class="n('title')" v-if="titlePosition === 'center'">
      <slot>{{ title }}</slot>
    </div>

    <div :class="n('right')">
      <div :class="n('title')" :style="{ paddingRight }" v-if="titlePosition === 'right'">
        <slot>{{ title }}</slot>
      </div>
      <slot name="right" />
    </div>
  </div>
</template>

<script lang="ts">
import { defineComponent, ref, Ref, onMounted, onUpdated } from 'vue'
import { props } from './props'
import { createNamespace } from '../../../components/utils/components'

const { n, classes } = createNamespace('app-bar')

export default defineComponent({
  name: 'VarAppBar',
  props,
  setup(props, { slots }) {
    const paddingLeft: Ref<number | undefined> = ref()
    const paddingRight: Ref<number | undefined> = ref()

    const computePadding = () => {
      paddingLeft.value = slots.left ? 0 : undefined
      paddingRight.value = slots.right ? 0 : undefined
    }

    onMounted(computePadding)
    onUpdated(computePadding)

    return {
      n,
      classes,
      paddingLeft,
      paddingRight,
    }
  },
})
</script>

<style lang="less">
@import '../../../components/styles/common';
@import '../../../components/styles/elevation';
@import './appBar';
</style>
