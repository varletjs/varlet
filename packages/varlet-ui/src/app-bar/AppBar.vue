<template>
  <div
    class="var-app-bar"
    :class="{ 'var-elevation--3': elevation }"
    :style="{
      background: color,
      color: textColor,
    }"
  >
    <div class="var-app-bar__left">
      <slot name="left" />
      <div class="var-app-bar__title" :style="{ paddingLeft }" v-if="titlePosition === 'left'">
        <slot>{{ title }}</slot>
      </div>
    </div>

    <div class="var-app-bar__title" v-if="titlePosition === 'center'">
      <slot>{{ title }}</slot>
    </div>

    <div class="var-app-bar__right">
      <div class="var-app-bar__title" :style="{ paddingRight }" v-if="titlePosition === 'right'">
        <slot>{{ title }}</slot>
      </div>
      <slot name="right" />
    </div>
  </div>
</template>

<script lang="ts">
import { defineComponent, ref, Ref, onMounted, onUpdated } from 'vue'
import { props } from './props'

export default defineComponent({
  name: 'VarAppBar',
  props,
  setup(props, { slots }) {
    const paddingLeft: Ref<number | undefined> = ref()
    const paddingRight: Ref<number | undefined> = ref()

    const computePadding = () => {
      paddingLeft.value = slots.left ? 0 : undefined
      paddingRight.value = slots.right ? 0 : undefined
    }

    onMounted(computePadding)
    onUpdated(computePadding)

    return {
      paddingLeft,
      paddingRight,
    }
  },
})
</script>

<style lang="less">
@import '../styles/common';
@import '../styles/elevation';
@import './appBar';
</style>
