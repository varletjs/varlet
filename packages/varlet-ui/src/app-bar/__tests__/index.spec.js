import AppBar from '..'
import VarAppBar from '../AppBar'
import { mount } from '@vue/test-utils'
import { createApp } from 'vue'

test('test app bar use', () => {
  const app = createApp({}).use(AppBar)
  expect(app.component(AppBar.name)).toBeTruthy()
})

describe('test app bar component props', () => {
  test('test app bar color', () => {
    const wrapper = mount(VarAppBar, {
      props: {
        color: 'red',
      },
    })

    expect(wrapper.find('.var-app-bar').attributes('style')).toContain('background: red;')
    wrapper.unmount()
  })

  test('test app bar round', () => {
    const wrapper = mount(VarAppBar, {
      props: {
        round: true,
      },
    })

    expect(wrapper.find('.var-app-bar').classes('var-app-bar--round')).toBeTruthy()
    wrapper.unmount()
  })

  test('test app bar textColor', () => {
    const wrapper = mount(VarAppBar, {
      props: {
        textColor: 'red',
      },
    })

    expect(wrapper.find('.var-app-bar').attributes('style')).toContain('color: red;')
    wrapper.unmount()
  })

  test('test app bar title', () => {
    const wrapper = mount(VarAppBar, {
      props: {
        title: 'text',
      },
    })

    expect(wrapper.text()).toBe('text')
    wrapper.unmount()
  })

  test('test app bar titlePosition', async () => {
    const wrapper = mount(VarAppBar, {
      props: {
        titlePosition: 'left',
        title: 'text',
      },
    })

    expect(wrapper.find('.var-app-bar__left').text()).toBe('text')

    await wrapper.setProps({
      titlePosition: 'right',
    })
    expect(wrapper.find('.var-app-bar__right').text()).toBe('text')

    await wrapper.setProps({
      titlePosition: 'center',
    })
    expect(wrapper.find('.var-app-bar__title').text()).toBe('text')

    wrapper.unmount()
  })

  test('test app bar elevation', async () => {
    const wrapper = mount(VarAppBar, {
      props: {
        elevation: true,
      },
    })

    expect(wrapper.classes().includes('var-elevation--3')).toBe(true)

    await wrapper.setProps({
      elevation: false,
    })
    expect(wrapper.classes().includes('var-elevation--3')).toBe(false)

    wrapper.unmount()
  })

  test('test app bar image and image-linear-gradient', () => {
    const wrapper = mount(VarAppBar, {
      props: {
        image: 'https://1.png',
        imageLinearGradient: 'to right top, rgba(29, 68, 147, 0.5) 0%, rgba(74, 198, 170, 0.9) 100%',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})

describe('test app bar slots', () => {
  test('test app bar default slot', () => {
    const wrapper = mount(VarAppBar, {
      slots: {
        default: () => 'This is default slot',
      },
    })

    expect(wrapper.text()).toBe('This is default slot')

    wrapper.unmount()
  })

  test('test app bar left slot', () => {
    const wrapper = mount(VarAppBar, {
      slots: {
        left: () => 'This is left slot',
      },
    })

    expect(wrapper.find('.var-app-bar__left').text()).toBe('This is left slot')

    wrapper.unmount()
  })

  test('test app bar right slot', () => {
    const wrapper = mount(VarAppBar, {
      slots: {
        right: () => 'This is right slot',
      },
    })

    expect(wrapper.find('.var-app-bar__right').text()).toBe('This is right slot')

    wrapper.unmount()
  })

  test('test app bar content slot', () => {
    const wrapper = mount(VarAppBar, {
      slots: {
        content: () => 'This is content slot',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()

    wrapper.unmount()
  })
})
