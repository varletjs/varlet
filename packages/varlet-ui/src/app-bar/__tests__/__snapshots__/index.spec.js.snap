// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test app bar example 1`] = `
"<div class=\\"app-type\\">基础导航栏</div>
<div class=\\"var-app-bar var-elevation--3\\">
  <div class=\\"var-app-bar__left\\">
    <div class=\\"var-app-bar__title\\">标题</div>
  </div>
  <!--v-if-->
  <div class=\\"var-app-bar__right\\">
    <!--v-if-->
  </div>
</div>
<div class=\\"app-type\\">自定义样式</div>
<div class=\\"var-app-bar var-elevation--3\\" style=\\"background: rgb(255, 159, 0);\\">
  <div class=\\"var-app-bar__left\\">
    <!--v-if-->
  </div>
  <div class=\\"var-app-bar__title\\">标题</div>
  <div class=\\"var-app-bar__right\\">
    <!--v-if-->
  </div>
</div>
<div class=\\"app-type\\">添加标题处插槽</div>
<div class=\\"var-app-bar var-elevation--3\\">
  <div class=\\"var-app-bar__left\\">
    <div class=\\"var-app-bar__title\\">从插槽处添加标题</div>
  </div>
  <!--v-if-->
  <div class=\\"var-app-bar__right\\">
    <!--v-if-->
  </div>
</div>
<div class=\\"app-type\\">添加左侧插槽</div>
<div class=\\"var-app-bar var-elevation--3\\">
  <div class=\\"var-app-bar__left\\"><button class=\\"var-button var--box var-button--normal var--inline-flex var-button--text-default var-button--text var-button--round\\" style=\\"color: rgb(255, 255, 255); background: transparent;\\">
      <!--v-if-->
      <div class=\\"var-button__content\\"><i class=\\"var-icon var-icon--set var-icon-chevron-left\\" style=\\"transition: transform 0ms; font-size: 24px;\\"></i></div>
    </button>
    <!--v-if-->
  </div>
  <div class=\\"var-app-bar__title\\">标题</div>
  <div class=\\"var-app-bar__right\\">
    <!--v-if-->
  </div>
</div>
<div class=\\"app-type\\">添加右侧插槽</div>
<div class=\\"var-app-bar var-elevation--3\\">
  <div class=\\"var-app-bar__left\\">
    <div class=\\"var-app-bar__title\\">标题</div>
  </div>
  <!--v-if-->
  <div class=\\"var-app-bar__right\\">
    <!--v-if--><button class=\\"var-button var--box var-button--normal var--inline-flex var-button--text-default var-button--text var-button--round\\" style=\\"color: rgb(255, 255, 255); background: transparent;\\">
      <!--v-if-->
      <div class=\\"var-button__content\\"><i class=\\"var-icon var-icon--set var-icon-magnify\\" style=\\"transition: transform 0ms; font-size: 24px;\\"></i></div>
    </button>
  </div>
</div>
<div class=\\"app-type\\">添加左右插槽</div>
<div class=\\"var-app-bar var-elevation--3\\">
  <div class=\\"var-app-bar__left\\"><button class=\\"var-button var--box var-button--normal var--inline-flex var-button--text-default var-button--text var-button--round\\" style=\\"color: rgb(255, 255, 255); background: transparent;\\">
      <!--v-if-->
      <div class=\\"var-button__content\\"><i class=\\"var-icon var-icon--set var-icon-chevron-left\\" style=\\"transition: transform 0ms; font-size: 24px;\\"></i></div>
    </button>
    <div class=\\"var-app-bar__title\\" style=\\"padding-left: 0px;\\">标题</div>
  </div>
  <!--v-if-->
  <div class=\\"var-app-bar__right\\">
    <!--v-if-->
    <div class=\\"var-menu\\"><button class=\\"var-button var--box var-button--normal var--inline-flex var-button--text-default var-button--text var-button--round\\" style=\\"color: rgb(255, 255, 255); background: transparent;\\">
        <!--v-if-->
        <div class=\\"var-button__content\\"><i class=\\"var-icon var-icon--set var-icon-menu\\" style=\\"transition: transform 0ms; font-size: 24px;\\"></i></div>
      </button>
      <!--teleport start-->
      <!--teleport end-->
    </div>
  </div>
</div>
<div class=\\"space\\"></div>"
`;

exports[`test app bar props 1`] = `
"<div class=\\"var-app-bar\\" style=\\"background: rgb(163, 185, 219); color: rgb(0, 0, 0);\\">
  <div class=\\"var-app-bar__left\\">leftSlot
    <!--v-if-->
  </div>
  <div class=\\"var-app-bar__title\\">test title</div>
  <div class=\\"var-app-bar__right\\">
    <!--v-if-->rightSlot
  </div>
</div>"
`;
